src/services/api/clinical-trials-search-api/getClinicalTrialDescription.js

```diff
@@ -4,23 +4,22 @@
  * @param {import("axios").AxiosInstance} client An axios instance with the correct baseURL
  * @param {object} query the cts query
  */
+
 export const getClinicalTrialDescription = async (client, query) => {
 	try {
 		const res = await client.get(`/trials/${query}`);
-		if (res.status === 200) {
-			return res.data;
-		} else {
-			// This condition will be hit for anything < 300.
+		if (res.status !== 200) {
+			// ie. Either a non-200 2xx status code or 3xx redirect
 			throw new Error(
 				`Unexpected status ${res.status} for fetching clinical trial description`
 			);
 		}
+		return res.data;
 	} catch (error) {
-		// This conditional will be hit for any status >= 300.
+		// 4xx or 5xx error
 		if (error.response) {
-			throw new Error(
-				`Unexpected status ${error.response.status} for fetching clinical trial description`
-			);
+			// We want to clarify the default message but propogate the extra data
+			error.message = `Unexpected status ${error.response.status} for fetching clinical trial description`;
```

Comment:
Why not throw a new exception here? setting the message here would overwrite the message here. You can include the inner exception https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#rethrowing_an_error_with_a_cause. (Passing the inner exception would be better than what had done)